plugins/woocommerce/includes/class-wc-ajax.php

```diff
@@ -2006,6 +2007,15 @@ public static function refund_line_items() {
 			}
 			foreach ( $line_item_qtys as $item_id => $qty ) {
 				$line_items[ $item_id ]['qty'] = max( $qty, 0 );
+				if ( isset( $order_line_items[ $item_id ] ) && $line_items[ $item_id ]['qty'] > $order_line_items[ $item_id ]->get_quantity() ) {
```

Comment:
Something we should consider here. Say we have an order like this:

```
Item    Qty    Cost
A         1     100
B         1      10
```

If the merchant tries to refund 2 units of Item B, this is triggered and we throw an exception. However, if they refund 1 unit, nothing prevents them from coming back via subsequent requests and repeating that process (ie, they could simply refund 1 unit 5 times, and over-refund that way).